### PR TITLE
feature/global_det_ee2

### PR DESCRIPTION
**Pull request by malloryprow**
## Pull Request Testing ##

The following was done:

1. Add module loads for PrgEnv-intel and intel, and removing /apps/prod/lmodules/intel/${intel_ver}
2. Remove hard-coded MET version in wave plotting configuration file
3. Replace "set +x" and "set -xa" with "set -x"
4. Remove "export LD_LIBRARY_PATH=/apps/dev/pmi-fix:$LD_LIBRARY_PATH"
5. Add python3 shebang to wave plotting scripts
6. Add plotting changes that were done in PR #234 
7. Add SENDDBN alerts for final plots tar file; set default in J-Jobs to SENDDBN=YES while setting to SENDDBN=NO in dev/drivers/scripts
8. Use compath.py to define COMOUT in J-Job
9. Remove RUN_ENVIR
10. Check for existence of file before copying to COMOUT in stats; had been trying to copy files that didn't exist causing failures (bug fix)
11. Move J-Jobs under jobs/
12. Match sorc directories to match their executable names
13. Print when a directory is made in a python script; no directories created without being logged
14. Add SENDMAIL; set default in J-Jobs to SENDMAIL=NO while setting to SENDMAIL=YES in dev/drivers/scripts
15.  Fix permissions and mode for ECMWF files
16. Remove hard-coded paths and add "debug", "clean", and "install" in make files
17. Remove KEEPDATA and SENDCOM from parm/evs_config files
18. Set KEEPDATA=YES in dev/drivers/scripts
19. Remove setting of COMOUT in plots dev/drivers/scripts
20. Increase atmos prep memory
21. Correctly define COMIN using compath.py
22. Use "export err=\$?; err_chk" in scripts/global_det/stats/exevs_global_det_wave_grid2obs_stats.sh

- [ ] Describe testing already performed for this Pull Request:</br>

> I tested prep, stats, and plots and compared with my parallel and the emc.vpppg parallel. 

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

**Set-up**
> 1. Clone my fork and checkout branch feature/global_det_ee2
> 2. ln -sf /lfs/h2/emc/vpppg/noscrub/emc.vpppg/verification/EVS_fix fix...or run sorc/build-fix-link.sh?
> 3. cd sorc; ./build.sh

**prep**
> 4. Run dev/drivers/scripts/global_det/prep/jevs_global_det_atmos_prep.sh (after 1400Z) and dev/drivers/scripts/global_det/prep/jevs_global_det_wave_prep.sh (after 1315Z).

**stats**
> 5. Run dev/drivers/scripts/global_det/stats/jevs_global_det_wave_gfs_grid2obs_stats.sh, dev/drivers/scripts/global_det/stats/jevs_global_det_atmos_jma_grid2grid_stats.sh, and dev/drivers/scripts/global_det/stats/jevs_global_det_atmos_ukmet_grid2obs_stats.sh. Point COMIN to the emc.vpppg parallel.

**plots**
> 6. Run dev/drivers/scripts/global_det/plots/jevs_global_det_wave_grid2obs_plots_31days.sh, dev/drivers/scripts/global_det/plots/jevs_global_det_atmos_headline_plots.sh, and dev/drivers/scripts/global_det/plots/jevs_global_det_atmos_grid2grid_means_plots_31days.sh. Point COMIN to the emc.vpppg parallel.

- [ ] Has the code been checked to ensure that no errors occur during the execution? **Yes**

- [ ] Do these updates/additions include sufficient testing updates? **Yes**

- [ ] Please complete this pull request review by **10/13/2023**.</br>

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

**Comment by PerryShafran-NOAA** [comment]
I have started work on this PR.

1) I am running the prep jobs.
2) I am running the various stats jobs.  I set COMIN to /lfs/h2/emc/vpppg/noscrub/emc.vpppg/$NET/$evs_ver.  I had to expressly write this out because of your use of COMROOT, which defines both COMIN and COMOUT in your scripts.  
3) I am running the various plots jobs.  I set COMIN to /lfs/h2/emc/vpppg/noscrub/emc.vpppg/$NET/$evs_ver

**Comment by PerryShafran-NOAA** [comment]
@malloryprow I am not entirely sure that the waves prep job worked.  It can't seem to find the buoys:  

/lfs/h2/emc/vpppg/noscrub/perry.shafran/pr279test/EVS/dev/drivers/scripts/global_det/prep/jevs_global_det_wave_prep_00.o83457666
```

662 + ls -l '/lfs/h1/ops/prod/dcom/20231004/validation_data/marine/buoy/*.txt'
ls: cannot access '/lfs/h1/ops/prod/dcom/20231004/validation_data/marine/buoy/*.txt': No such file or directory
62 + nbdc_txt_ncount=0
62 + [[ 0 -eq 0 ]]
62 + '[' YES = YES ']'
62 + export 'subject=NDBC Data Missing for EVS global_det'
62 + subject='NDBC Data Missing for EVS global_det'
62 + echo 'Warning: No NDBC data was available for valid date '
62 + echo 'Missing files are located at /lfs/h1/ops/prod/dcom/20231004/validation_data/marine/buoy'
62 + echo 'Job ID: jevs_global_det_wave_prep_00.83457666.cbqs01'
63 + cat mailmsg
63 + mail -s 'NDBC Data Missing for EVS global_det' alicia.bentley@noaa.gov,mallory.row@noaa.gov
63 + echo ' '
```

**Comment by malloryprow** [comment]
Ah yeah they don't come in until a little later this morning. The cron runs at 1315 I think?

**Comment by PerryShafran-NOAA** [comment]
@malloryprow @AliciaBentley-NOAA @ShelleyMelchior-NOAA I think there are issues in your setup.

I set COMIN to /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0

I cannot change your definition of COMROOT to emc.vpppg, because your COMOUT also depends on COMROOT.  This is a problem, because I cannot *write* to emc.vpppg in my testing .  

Your various COMINs cannot use -o in the J-job; that is an NCO requirement.  Therefore, some changes will need to be made I think in order to get this to work.  

You need to set all your COMINs in the J-job to look like this (no -o):
export COMINukmet=${COMINukmet:-$(compath.py ${envir}/${com}/$NET/$evs_ver/prep/$COMPONENT/$RUN)}

And then set COMINukmet in your dev driver script to look like this:

export COMINukmet=$COMIN/prep/$COMPONENT/$RUN

We will need these changes for the code to a) comply with NCO rules, and to b) work.

**Comment by malloryprow** [comment]
Okay, well please don't do any more testing until I can get these changes done and test them myself. I thought it would be as simple as setting COMIN to /lfs/h2/emc/vpppg/noscrub/emc.vpppg/$NET/$evs_ver rather than changing the whole COMROOT variable.

**Comment by malloryprow** [comment]
So I made the COMIN compath.py changes...now I'm getting 

> compath.py evs/v1.0
> Could not find evs/v1.0
> 1 + export COMIN=
> 1 + COMIN=

so now does COMIN need to be defined in the drivers? That seems to be the only thing that makes sense.

**Comment by PerryShafran-NOAA** [comment]
Yes.  COMIN should be defined in the driver.

**Comment by malloryprow** [comment]
Also are we sure that COMIN[model] needs to go back into the drivers? I just removed this in a PR a few weeks ago (https://github.com/NOAA-EMC/EVS/pull/260). I don't think I would have removed those unless it was required that they NOT be in the drivers.

**Comment by malloryprow** [comment]
Do all COMIN paths have to be defined with compath.py or can we use COMIN?

**Comment by AliciaBentley-NOAA** [comment]
@malloryprow I'm going to send an email to Steven Earle about this and CC you. Stand by! :)

**Comment by AliciaBentley-NOAA** [comment]
@malloryprow 
In both 1) JEVS_GLOBAL_DET_ATMOS_LONG_TERM_STATS and 2) JEVS_GLOBAL_DET_ATMOS_LONG_TERM_PLOTS, can you please update these two "VDATE" references to use PDY and not just date? Thanks!
```
export VDATEYYYY=${VDATEYYYY:-$(date -d "1 month ago" '+%Y')}
export VDATEmm=${VDATEmm:-$(date -d "1 month ago" '+%m')}
```

**Comment by malloryprow** [comment]
JEVS_GLOBAL_DET_ATMOS_LONG_TERM_STATS and JEVS_GLOBAL_DET_ATMOS_LONG_TERM_PLOTS won't be delivered to NCO. Is it okay to leave them? I'm not sure how to use PDY to get the previous month.

**Comment by AliciaBentley-NOAA** [comment]
@malloryprow Ah, yes. The j-jobs shouldn't be given to NCO either, not just the dev driver scripts! Can you please remind me of the exevs_ scripts that should be removed too? I'll write them all out in the EE2 checklist (like you recently suggested in a comment in that document). 

@PerryShafran-NOAA @ShelleyMelchior-NOAA Please be aware that we'd like the following "long_term" scripts from Mallory removed from the EVS parallel.

**Comment by AliciaBentley-NOAA** [comment]
@malloryprow Please change instances of "set -xa" to "set -x" in your global_det PR. I see "set -xa" in [JEVS_GLOBAL_DET_PLOTS](https://github.com/NOAA-EMC/EVS/pull/279/files#diff-ea108c16281a35c4edaf58141af3108448b938659967043e0797ab5f60435810), but it may be in other places too! Thanks for checking!

**Comment by malloryprow** [comment]
All the scripts to be removed that pertain to the global_det long-term:

1. /dev/drivers/scripts/global_det/stats/jevs_global_det_atmos_long_term_stats.sh
2. dev/drivers/scripts/global_det/plots/jevs_global_det_atmos_long_term_plots.sh
3. jobs/global_det/stats/JEVS_GLOBAL_DET_ATMOS_LONG_TERM_STATS
4. jobs/global_det/plots/JEVS_GLOBAL_DET_ATMOS_LONG_TERM_PLOTS
5. scripts/global_det/stats/exevs_global_det_atmos_long_term_stats.sh
6. scripts/global_det/plots/exevs_global_det_atmos_long_term_plots.sh
7. ush/global_det/global_det_atmos_plots_long_term.py
8. ush/global_det/global_det_atmos_plots_long_term_annual_mean.py
9. ush/global_det/global_det_atmos_plots_long_term_lead_by_date.py
10. ush/global_det/global_det_atmos_plots_long_term_time_series.py
11. ush/global_det/global_det_atmos_plots_long_term_time_series_diff.py
12. ush/global_det/global_det_atmos_plots_long_term_time_series_multifhr.py
13. ush/global_det/global_det_atmos_plots_long_term_useful_forecast_days.py
14. ush/global_det/global_det_atmos_stats_long_term.py

File: jobs/JEVS_GLOBAL_DET_PLOTS [hunk 1]
```diff
@@ -44,7 +44,7 @@ export VDATE_END=${VDATE_END:-$PDYm1}
 ####################################
 export SENDECF=${SENDECF:-YES}
 export SENDCOM=${SENDCOM:-YES}
-export SENDDBN=${SENDDBN:-NO}
+export SENDDBN=${SENDDBN:-YES}
```
Comment (AliciaBentley-NOAA) [inline jobs/JEVS_GLOBAL_DET_PLOTS:47]: @malloryprow Can you please add _export SENDMAIL=${SENDMAIL:-NO}_ into your plots J-job, just incase it's needed someday?

Comment (malloryprow) [reply jobs/JEVS_GLOBAL_DET_PLOTS:47]: Resolved in https://github.com/NOAA-EMC/EVS/pull/279/commits/e9e2f630f1ac945d51a55c3f3b9c6669fd04f125

**Comment by AliciaBentley-NOAA** [comment]
@malloryprow I have completed my review of your global_det EE2 PR against the EE2 checklist! Everything looks great with the minor changes that you've already made. 

@PerryShafran-NOAA @ShelleyMelchior-NOAA We're still waiting to hear back from Steve Earle about COMIN/COMOUT, but I support testing this PR whenever you are comfortable with the status of COMIN/COMOUT. Thanks!

**Comment by malloryprow** [comment]
Okay, I fixed up the COMIN paths. I haven't tested the changes though. The wave prep cron starts running at 1315Z and the atmos prep cron run at 1400Z.

**Comment by malloryprow** [comment]
This is ready for testing. Let's start with the prep jobs first.

**Comment by PerryShafran-NOAA** [comment]
OK, prep jobs are underway.

**Comment by PerryShafran-NOAA** [comment]
The waves prep step is complete.  See:  

.o file:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr279test/EVS/dev/drivers/scripts/global_det/prep/jevs_global_det_wave_prep_00.o83560774

output:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/prep/global_det/wave.20231004

**Comment by malloryprow** [comment]
:heavy_check_mark: **prep tests are good!** 

Output from jevs_global_det_atmos_prep.sh and jevs_global_det_wave_prep.sh are the same as the parallel!

We can move on to the stats tests.

**Comment by PerryShafran-NOAA** [comment]
OK, I just changed COMIN to point to emc.vpppg.  I did not change COMROOT.  I am running the requested prep jobs.

**Comment by malloryprow** [comment]
HOMEevs needs to point to lfs/h2/emc/vpppg/noscrub/perry.shafran/pr279test/EVS in /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr279test/EVS/dev/drivers/scripts/global_det/stats/jebal_det_atmos_jma_grid2grid_stats.sh

**Comment by PerryShafran-NOAA** [comment]
Darn, missed that.  Fixed, and re-submitted.

How do the other two jobs look?

**Comment by malloryprow** [comment]
Perfect! The final stat files match the emc.vpppg parallel.

**Comment by malloryprow** [comment]
:heavy_check_mark: **stats tests are good!**

Output from jevs_global_det_wave_gfs_grid2obs_stats.sh, jevs_global_det_atmos_jma_grid2grid_stats.sh, and jevs_global_det_atmos_ukmet_grid2obs_stats.sh are the same as the parallel!

We can move on to the plots tests.

**Comment by PerryShafran-NOAA** [comment]
Plots testing is underway.

**Comment by malloryprow** [comment]
:heavy_check_mark: **plots tests are good!**

Output from jevs_global_det_wave_grid2obs_plots_31days.sh, jevs_global_det_atmos_headline_plots.sh, and jevs_global_det_atmos_grid2grid_means_plots_31days.sh match the plots output that I ran.

Merge away!

**Review by ShelleyMelchior-NOAA — APPROVED** [review]
All testing produced expected results. This PR is ready for merge.

**Pull request merged**